Add support for Rails TZ handling

ares opened the pull request:
@adamruzicka mind to take a look? Not sure how to properly test this since ActiveSupport::TimeZone is not a dependency.

adamruzicka commented on lib/scoped_search/definition.rb:
```diff
@@ -278,6 +278,8 @@ def default_fields_for(value, operator = nil)
     # Try to parse a string as a datetime.
     # Supported formats are Today, Yesterday, Sunday, '1 day ago', '2 hours ago', '3 months ago', '4 weeks from now', 'Jan 23, 2004'
     # And many more formats that are documented in Ruby DateTime API Doc.
+    # In case Time responds to #zone, we know this is Rails environment and we can use Time.zone.parse. The benefir is that the
```
`s/benefir/benefit`

adamruzicka commented on lib/scoped_search/definition.rb:
```diff
@@ -286,7 +288,11 @@ def parse_temporal(value)
+      if Time.respond_to?(:zone)
+        Time.zone.parse(value)
+      else
+        DateTime.parse(value, true) rescue nil
```
We should rescue the `Time.zone` branch too

ares replied:
in case of wrong format it does not raise exception but returns nil

adamruzicka replied:
Not always
```ruby
pry(main)> Time.zone.parse('123:15:00')
ArgumentError: argument out of range
```

ares replied:
interesting, updated

adamruzicka commented on lib/scoped_search/definition.rb:
```diff
@@ -286,7 +288,11 @@ def parse_temporal(value)
       return (eval($1.strip.gsub(/from\s+now/i,'from_now').gsub(/\s+/,'.').downcase)).to_date     if value =~ /\A\s*(\d+\s+\b(?:days?|weeks?|months?|years?)\b\s+\bfrom\s+now)\b\s*\z/i
-      DateTime.parse(value, true) rescue nil
+      if Time.respond_to?(:zone)
+        Time.zone.parse(value)
```
Maybe add `.to_datetime` at the end to always return `DateTime`?

ares commented:
travis is no longer triggered for PRs, @wvanbergen do you happen to know why? Anyway, this should fix all comments and add testing with active support

adamruzicka commented:
Looks good to me, waiting for Travis to confirm

ares commented:
Since I don't have admin permissions over the repo, I can't look at the hooks configuration. @wvanbergen could you please take a look?

wvanbergen commented:
Looks like it was deactivated on Travis's end. I think I have re-enabled it. Can you try by pushing something to this branch?

ares commented:
Thanks @wvanbergen, it seems travis is now running again!

adamruzicka commented:
Travis is running but doesn't like the changes

ares commented:
fixed the travis definition, it seems now that the change has some negative effect, I'll need to look into more details, seems that the change affected even gemsets without active support

ares commented:
Tests are green now, I updated travis configuration too and removed some old rubies to make the test matrix smaller. I think this is ready to go in. Btw tests revealed an issue if Time.zone is nil (no timezone set) so fixed that as well.

ares commented:
@wvanbergen any concern with old rubies drop from travis matrix? I think we really don't need them anymore

wvanbergen commented:
Can we still support ruby 2.3.7? That's the version macOS ships with.

ares commented:
Fair enough, in fact RHEL 7 still ships Ruby 2.0 so I kept old version. The matrix is now bigger but travis seems to handle it fine. Let's see if all turn green now

ares commented:
OK, all green now including old rubies :-)

ares commented:
could this be merged pls?

adamruzicka commented:
Thanks @ares !

ares commented:
Thank you!
